test/type_check/internals/parser_test.exs

```diff
@@ -1,6 +1,6 @@
 defmodule TypeCheck.Internals.ParserTest do
   @moduledoc false
-  use ExUnit.Case, async: true
+  use ExUnit.Case
```

Comment:
Interesting. I'll do some checking afterwards to see whether we can go back to `async`, because there is no clear reason why it should not be possible.

Reply:
Yeah, I agree in general and I'm opening an issue in Elixir. In this case though, I'm not sure that defining the same module "simultaneously" is the way to go...

Reply:
I recommend waiting with that; I think that there is a 90% chance that the issue is in this library rather than in Elixir v1.14. :sweat_smile:

Reply:
FYI, I opened https://github.com/elixir-lang/elixir/issues/12189

Reply:
You managed to extract a very nice concise counter-example, so maybe it is a problem there? let's see what the Elixir team thinks.